server/app/routes/character/index.js

```diff
@@ -30,14 +33,19 @@ router.param('id', (req, res, next, id) => {
 
 // update a character
 router.put('/:id', (req, res, next) => {
-	req.requestedCharacter.update(req.body)
-	.then(updatedCharacter => res.json(updatedCharacter))
-	.catch(next);
+	if(!req.user) res.sendStatus(401)
+	else if (req.user.isAdmin || req.user.equals(req.requestedCharacter)) {
+		req.requestedCharacter.update(req.body)
+		.then(updatedCharacter => res.json(updatedCharacter))
+		.catch(next);
+	} else res.sendStatus(401)
 });
 
 // delete a character
 router.delete("/:id", (req, res, next) => {
-	req.requestedCharacter.remove()
-	.then(() => res.sendStatus(204))
-	.catch(next);
+	if(req.user || req.user.isAdmin) {
```

Comment:
do we need "|| req.user.isAdmin" ?  if it's enough to be a user, right?  (and an admin is a user)